code/game/mecha/mecha.dm

```diff
@@ -878,6 +878,9 @@
 		return
 	if(!ishuman(user)) // no silicons or drones in mechas.
 		return
+	if(HAS_TRAIT(user, TRAIT_PRIMITIVE)) //no lavalizards either.
+		to_chat(user, "<span class='warning'>The knowledge to use this device eludes you!</span>")

```

Comment:
this is a bit awkwardly worded

also, RIP ashwalkers finding and using the abandoned lavaland ripley that sometimes spawns somewhere in lavaland

Reply:
Also, I'm not really sure about specifically saying "The knowledge to use this device eludes you!" based on race alone. If someone uses a pride mirror, envy's knife, a mutation toxin, a magic mirror, a changeling's transform ability, etc. to become an ash walker, why would lose their knowledge of how to pilot a mech?

If anything, you should be checking specifically for the ashwalker antag datum, although honestly, if an ash walker manages to find and hijack a mech that hasn't been DNA locked, I say that they've probably earned the right to use it.  Like, if you put a heavy laser cannon in a mech, didn't DNA lock it, and left it in a place where an ash walker could just walk up to it and climb in (or if you left Robotics unsupervised and unattended long enough for an ash walker to create their own mech (and modules for it)), it's entirely YOUR fault that an ash walker got access to a ranged weapon in the form of mech-mounted guns.

Note that wearing a space suit or the like already allows ashwalkers to cover their feetsies, so acquiring an immunity to shardstuns isn't anything particularly special. Note that golems, which also aren't supposed to be able to use most guns, can still pilot mechs and fire mech-mounted guns, and nobody seems to be having a problem with that (even though free golems have the means to create mechs on their own instead of stealing them). Also note that, IIRC, ash walkers can raid the free golem ship to get PKA trigger guard adapters so that they can fire PKAs or use envy's knife to change their race to, say, a skeleton, so even ash walkers getting access to ranged weapons isn't all that special either.

Reply:
Ash Walkers being able to raid the golem ship mean that it's not always the crew's fault that an Ash Walker gets a Durand.

Anyway, the issue of the Pride's mirror has been noted and I'll soon change it to a trait added during the creation of a new Ash Walker via the spawn egg (as a "roundstart" category trait). This means it'll no longer apply to anyone that becomes an Ash Walker by other means and any Ash Walker using the mirror won't suddenly learn mech piloting.